components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -355,6 +355,20 @@ public String streamingPredict(int tenantId, String userName, long modelId, Stri
             throw new MLModelHandlerException(msg);
         }
 
+        // Validate numerical feature type in predict dataset
+        for (Feature feature: builtModel.getFeatures()) {
+            if (feature.getType().equals(FeatureType.NUMERICAL)) {
+                int actualIndex = builtModel.getNewToOldIndicesList().indexOf(feature.getIndex());
+                for (String[] dataPoint: data) {
```

Comment:
In worst case, this would not perform well. Can we first try to find the list of numerical features and then check whether the data contains invalid values?